### DISCUSSION TITLE
how to change the image pull policy of the workflow step pod

### DISCUSSION THREAD

**Discussion by zsyet2018** [argoproj/argo-workflows#7347] [Q&A] [answered]
for example, the workflow step pod like the following:

apiVersion: v1
kind: Pod
metadata:
  ...
  ...
  name: hello-world-2
  namespace: argo
  ...
  ...
spec:
  containers:
  - command:
    - argoexec
    - wait
    - --loglevel
    - info
    **image: quay.io/argoproj/argoexec:latest
    imagePullPolicy: Always**
    name: wait
...
...

is it possible to change imagePullPolicy? it's default value is "Always", and how to change it

**Comment by terrytangyuan** [comment, marked as answer]
You can change the main container's spec via podSpecPatch in workflowSpec or add your executor container spec in controller's configmap.  See #7342 too.
